internal/app/api/server/private/modules.go

```diff
@@ -4038,7 +4121,8 @@ func (s *ModuleService) PatchModuleVersionFile(c *gin.Context) {
 		return
 	}
 
-	if s3, err = storage.NewS3(nil); err != nil {
+	s3, err := storage.NewS3(nil)
```

Comment:
(not for this PR) it seems like here we just get connection to a global S3 which can be fetched from cache (or be part of injected dependencies)

Reply:
To keep changes small I avoided this refactoring, but did method in connector. Let's do it in another PR